specs/agents/tracing-spans.md

```diff
@@ -33,6 +33,16 @@ Each span will have a `name`, which is a descriptive, low-cardinality string.
 
 If a span is created without a valid `name`, the string `"unnamed"` SHOULD be used.
 
+### Span `sync`
+
+Span execution within a transaction or span can be synchronous (the caller waits for completion), or asynchronous (the caller does not wait
+for completion).
```

Comment:
Maybe here we should clarify that the agents should only be able to capture the (a)synchronous "nature" of a specific span, for example when a synchronous or asynchronous API is used.

It might sound obvious, but if an asynchronous span is (mis)used in a synchronous way this can't easily be detected by agents at runtime, so from the user perspective it's only in the "shape" of the trace and the relative position of the blocking or async spans that we can infer if that's an issue or not.

Few examples:
- an asynchronous API is used in a blocking way: having the `async` span making the parent span/transaction longer indicates that refactoring code could help.
- starting multiple async tasks from a single span/transaction and then awaiting their completion in a  blocking way would not be considered a problem because the async tasks are started in parallel. However, doing the same with a single span would make it fall in the "async API used in a blocking way" described above.